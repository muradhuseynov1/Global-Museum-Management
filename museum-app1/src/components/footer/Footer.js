import React from 'react';
import { Box, Typography, Container, useTheme } from '@mui/material';
import thesisIcon from '../../assets/thesis_icon1.png';

const Footer = () => {
  const theme = useTheme();

  const footerStyle = {
    backgroundColor: '#123456', 
    color: theme.palette.getContrastText('#123456'), 
    padding: theme.spacing(2),
    marginTop: 'auto', 
    zIndex: theme.zIndex.appBar
  };

  const contentStyle = {
    display: 'flex',
    alignItems: 'center', 
    justifyContent: 'center', 
    gap: theme.spacing(2)
  };

  return (
    <Box component="footer" sx={footerStyle}>
      <Container maxWidth="lg">
        <Box sx={contentStyle}>
          <img src={thesisIcon} alt="Logo" style={{ width: '50px', height: '50px' }} />
          <Box>
            <Typography variant="subtitle2" color="inherit" gutterBottom>
              Global Museum Management
            </Typography>
            <Typography variant="caption" color="inherit">
              Â© {new Date().getFullYear()} Global Museum Management. All rights reserved.
            </Typography>
          </Box>
        </Box>
      </Container>
    </Box>
  );
};

export default Footer;
