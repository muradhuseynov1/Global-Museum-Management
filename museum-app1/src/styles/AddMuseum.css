.react-datepicker__day--highlighted {
    background-color: #ffdd00; 
    color: #000; 
  }  

  html, body, #root {
    margin: 0;
    padding: 0;
    width: 100%;
    height: 100%;
  }

  
  .addMuseumBackground {
  background-color: #123456;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.addMuseumContainer {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 16px;
}

.addMuseumPaper {
  background-color: white;
  border-radius: 8px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.addMuseumStepper {
  width: 100%;
}

.addMuseumFormBody {
  overflow-y: auto; 
  padding: 16px;
  display: flex; 
  justify-content: center; 
  align-items: center; 
  flex-direction: column; 
}

.addMuseumFormFooter {
  display: flex;
  width: 100%;
  justify-content: space-between;
  padding: 16px;
  box-sizing: border-box;
}

.responsive-calendar {
  max-width: 100%; 
}

.react-datepicker {
  font-size: 0.875rem; 
}

.react-datepicker__header {
  background-color: #123456; 
  color: white; 
}

.react-datepicker__navigation {
  top: 12px; 
  border-color: white; 
}

.react-datepicker__navigation-icon::before {
  border-color: white; 
}

.react-datepicker__header__dropdown,
.react-datepicker__day, 
.react-datepicker__time-container {
  padding: 4px; 
  min-width: 0; 
}

.react-datepicker__current-month,
.react-datepicker-time__header,
.react-datepicker-year-header {
  color: white;
}

.react-datepicker__day-name {
  color: white;
}

.react-datepicker__month-dropdown-container,
.react-datepicker__year-dropdown-container,
.react-datepicker__month-year-dropdown-container {
  select {
    background-color: white; 
    color: #123456; 
    border-radius: 4px; 
    padding: 4px; 
    border: 1px solid #ccc; 
  }
}

.react-datepicker__navigation--previous,
.react-datepicker__navigation--next {
  top: 10px; 
}

.custom-datepicker-wrapper {
  display: flex;
  justify-content: center; 
  width: 100%; 
}
