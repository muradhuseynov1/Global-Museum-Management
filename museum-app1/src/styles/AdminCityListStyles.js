import websiteBackground from '../assets/website_background.jpg';

const AdminCityListStyles = {
    background: {
        display: 'flex',
        flexDirection: 'column',
        minHeight: '100vh',
        backgroundImage: `url(${websiteBackground})`,
        backgroundSize: 'cover',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
    },
};

export default AdminCityListStyles;
