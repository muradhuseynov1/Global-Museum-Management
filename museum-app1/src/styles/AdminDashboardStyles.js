import websiteBackground from '../assets/website_background.jpg';

const AdminDashboardStyles = {
    dashboardBackground: {
        display: 'flex',
        flexDirection: 'column',
        minHeight: '100vh',
        backgroundImage: `url(${websiteBackground})`,
        backgroundSize: 'cover',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
    },
};

export default AdminDashboardStyles;
