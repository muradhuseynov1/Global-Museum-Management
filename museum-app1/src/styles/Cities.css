.citiesAppBar.MuiAppBar-colorPrimary {
  background-color: #123456; 
}
