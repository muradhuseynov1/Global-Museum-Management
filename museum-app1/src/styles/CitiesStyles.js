import websiteBackground from '../assets/website_background.jpg';

const CitiesStyles = {
    root: {
      display: 'flex',
      flexDirection: 'column',
      minHeight: '100vh',
      backgroundImage: `url(${websiteBackground})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'center',
    }
};

export default CitiesStyles;