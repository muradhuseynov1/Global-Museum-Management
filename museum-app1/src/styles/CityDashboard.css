.dashboardAppBar {
    background-color: #123456; 
  }
  
.dashboardContainer {
margin-top: 20px;
}

.dashboardTitle {
margin-bottom: 20px;
color: #123456; 
}

.museumGrid {
margin-top: 20px;
}

.museumCard {
    display: flex;
    flex-direction: column;
    transition: transform 0.15s ease-in-out;
    border-radius: 8px; 
    overflow: hidden; 
    box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1); 
}
  
.cardMedia {
height: 140px;
background-size: cover;
background-position: center; 
}

.museumCardContent {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: center; 
    background-color: #fff;
    padding: 16px; 
    min-height: 100px !important; 
    overflow: hidden; 
}

  
.iconContainer {
    position: absolute;
    top: 10px;
    right: 10px;
    display: flex; 
    gap: 8px; 
    visibility: hidden; 
}

.museumCard:hover .iconContainer {
    visibility: visible; 
}
  
.editIconButton, .deleteIconButton {
    background-color: rgba(255, 255, 255, 1) !important; 
    color: #1976d2; 
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5); 
    border-radius: 50%; 
    transition: transform 0.2s ease-in-out, box-shadow 0.3s;
    padding: 8px; 
}
  
.editIconButton:hover, .deleteIconButton:hover {
    transform: scale(1.2);
    background-color: rgba(255, 255, 255, 1); 
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.7); 
}

.museumName {
    white-space: nowrap; 
    overflow: hidden; 
    text-overflow: ellipsis; 
}

.toolbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
}
  
.toolbarSearch {
    flex-grow: 1;
    margin: 0 20px; 
    max-width: 600px; 
}
  
.toolbarIcons {
    display: flex;
    align-items: center;
    gap: 12px; 
}
  

.dashboardAppBar.MuiAppBar-root {
    background-color: #123456; 
}
  