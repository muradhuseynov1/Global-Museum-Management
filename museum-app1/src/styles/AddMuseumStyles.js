import { createTheme } from '@mui/material/styles';
import { styled } from '@mui/material/styles';
import { Box, Container, Paper, Stepper } from '@mui/material';

export const theme = createTheme({
  palette: {
    primary: {
      main: '#123456',
      light: '#345678',
    },
    common: {
      white: '#ffffff',
    },
  },
});

export const StyledBox = styled(Box)(({ theme }) => ({
  backgroundColor: '#123456',
  minHeight: '100vh',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
}));

export const StyledContainer = styled(Container)(({ theme }) => ({
  width: '100%',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  padding: theme.spacing(2),
}));

export const StyledPaper = styled(Paper)(({ theme }) => ({
  backgroundColor: 'white',
  borderRadius: theme.shape.borderRadius,
  width: '100%',
  padding: theme.spacing(3),
  boxSizing: 'border-box',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
}));

export const StyledStepper = styled(Stepper)(({ theme }) => ({
  paddingTop: theme.spacing(3),
  paddingBottom: theme.spacing(3),
  width: '100%',
}));

export const StyledForm = styled('form')({
  width: '100%',
});

export const StyledButtonsBox = styled(Box)(({ theme }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  width: '100%',
  paddingTop: theme.spacing(2),
  paddingBottom: theme.spacing(2),
}));
