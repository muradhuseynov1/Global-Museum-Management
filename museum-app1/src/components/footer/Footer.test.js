import { render, screen } from '@testing-library/react';
import Footer from './Footer';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import thesisIcon from '../../assets/thesis_icon1.png';

describe('Footer', () => {
  test('renders content and applies styles correctly', () => {
    const theme = createTheme();
    render(
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>
    );

    expect(screen.getByText('Global Museum Management')).toBeInTheDocument();
    expect(screen.getByText(`Â© ${new Date().getFullYear()} Global Museum Management. All rights reserved.`)).toBeInTheDocument();
    expect(screen.getByAltText('Logo')).toHaveAttribute('src', thesisIcon);
  });

  it('should match snapshot', () => {
    const theme = createTheme();
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
