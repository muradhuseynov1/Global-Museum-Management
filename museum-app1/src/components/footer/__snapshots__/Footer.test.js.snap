// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="MuiBox-root css-1lmadg6"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-1mz6ms4"
      >
        <img
          alt="Logo"
          src="thesis_icon1.png"
          style="width: 50px; height: 50px;"
        />
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-74i0q5-MuiTypography-root"
          >
            Global Museum Management
          </h6>
          <span
            class="MuiTypography-root MuiTypography-caption css-ffdaoj-MuiTypography-root"
          >
            Â© 2024 Global Museum Management. All rights reserved.
          </span>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
